lib/webmock/fixtures/manager.rb

```diff
@@ -3,6 +3,103 @@ module WebMock
+      # @classmethod
+      # @param subclass [Class] the class instance which is subclassing `Manager`
+      def self.inherited(subclass)
+        subclass.instance_variable_set(:@registered_fixtures, {})
+      end
+
+      # Preload a web fixture which can then be started via `Manager.run`
+      # @classmethod
+      # @param name [Symbol] the name to register the fixture as
+      # @param verb [Symbol] symbol method to mock (e.g. `:get`, `:post`)
+      # @param pattern [Regexp|String] URI pattern to match for this mock
+      # @param response [String|File|Lambda|Hash] the response, this is the same as what would be supplied
+      #   to `WebMock::RequestStub#to_return` please see https://github.com/bblimke/webmock for examples
+      def self.register_fixture(name, verb, pattern, response)
+        @registered_fixtures[name] = {
+          :pattern => pattern,
+          :verb => verb,
+          :response => response,
+        }
+      end
+
+      # Retrieve a hash of registered fixtures available to this Manager
+      # @classmethod
+      # @return [Hash] a hash mapping the registered name to a hash of properties
+      #   e.g. `{ :get_example => { :pattern => %r{example.org}, :verb => :get, :response => "Hello World" } }`
+      def self.registered_fixtures
+        return @registered_fixtures
+      end
+
+      # Reset this manager by removing all previously registered fixtures
+      # @classmethod
+      def self.reset!
+        @registered_fixtures = {}
+      end
+
+      # Preload a web fixture from a file which can then be started via `Manager.run`
+      # @classmethod
+      # @param name [Symbol] the name to register the fixture as
+      # @param verb [Symbol] symbol method to mock (e.g. `:get`, `:post`)
+      # @param pattern [Regexp|String] URI pattern to match for this mock
+      # @param file_name [String] the name of the file to load the fixture response from
+      def self.load_fixture_file(name, verb, pattern, file_name)
+        # Read the contents from the file and use as the response
+        # https://github.com/bblimke/webmock/tree/v1.22.1#replaying-raw-responses-recorded-with-curl--is
+        register_fixture(name, verb, pattern, File.new(file_name).read)
+      end
+
+      # Start the named preloaded web fixtures
+      # @classmethod
+      # @param fixture_names [Array] the symbol names of fixtures to load and start
+      # @param fixture_names[*] [Symbol] the symbol name of the the fixture to load
+      # @return [Manager] an instance of Manager which has properties set for each started fixture
+      # @raise [KeyError] if the fixture has not been preloaded via `register_fixture` or `load_fixture_file`
+      def self.run(fixture_names)
+        # Create a new instance to store started mocks on
+        manager = new()
+        fixture_names.each do |fixture_name|
+          # DEV: `fetch` will fail if the key does not exist
+          options = @registered_fixtures.fetch(fixture_name)
```

Comment:
While a `KeyError` is a great, it isn't always that human friendly. Maybe use a `if @registered_fixtures.nil?; raise('Human friendly error message about not finding fixture_name, please verify no typos or use `register_fixture`. kthxbai')`